pkg/dependencies/factory.go

```diff
@@ -116,6 +117,7 @@ func ForSpec(ctx context.Context, clusterSpec *cluster.Spec) *Factory {
 type Factory struct {
 	executablesConfig        *executablesConfig
 	registryMirror           string
+	registryAuth             bool
```

Comment:
I think we probably want to consolidate these under a single configuration object.